haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -388,13 +402,13 @@ refsFromRenamed ctx declAlts (hsGroup, _, _, _) =
         -- Basic variable at the leaves of type trees.
         -- Not only "real" type variables, but also type-level terms like
         -- specific types, type aliases etc.
-        HsTyVar n -> give ctx (nameLocToRef n Ref l)
+        HsTyVar n -> give ctx (nameLocToRef (unLoc n) Ref l)
```

Comment:
Good idea, I have made the mayUnLoc but I haven't applied it in all relevant spots yet.